packages/auth/src/core/credentials/email.test.ts

```diff
@@ -237,59 +279,39 @@ describe('core/credentials/email', () => {
             400
           );
 
-          // Second call with a valid recaptcha token (captchaResp) should succeed
-          mockEndpointWithParams(
-            Endpoint.SIGN_IN_WITH_PASSWORD,
-            {
-              captchaResponse: 'recaptcha-response',
-              clientType: RecaptchaClientType.WEB,
-              email: 'some-email',
-              password: 'some-password',
-              recaptchaVersion: RecaptchaVersion.ENTERPRISE,
-              returnSecureToken: true
-            },
-            {
-              idToken: 'id-token',
-              refreshToken: 'refresh-token',
-              expiresIn: '1234',
-              localId: serverUser.localId!
-            }
+          // First call without recaptcha token should fail with MISSING_RECAPTCHA_TOKEN error
+          // Though the internal implementation retries with a reCAPTCHA Enterprise token, the second call will fail in this test.
+          // This is because our endpoint mock does not support returning different responses based on different request body params.
+          // TODO(renkelvin) - refactor this once we expose a mockEndpointWithBodyParams or similar method.
```

Comment:
What does this method do?

Reply:
You mean what SHOULD mockEndpointWithBodyParams do? We need a method that will send a different response based on a different request body.

mockEndpointWithParams - is an existing method which sends the specified response, for the request with the given URL Params. But our sign up/sign in are POST methods, so they never match the url param or the mock.